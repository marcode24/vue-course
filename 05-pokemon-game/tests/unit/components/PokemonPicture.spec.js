import { shallowMount } from '@vue/test-utils'
import PokemonPicture from '@/components/PokemonPicture'

describe('PokemonPicture component', () => {

  test('Debe hacer match con el snapshot', () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 12,
        showPokemon: false
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('Debe de mostrar la imagen oculta y el pokemon 100', () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 100,
        showPokemon: false
      }
    })
    const [ img1, img2 ] = wrapper.findAll('img')
    expect(img1.exists()).toBeTruthy()
    expect(img2).toBe(undefined)
    expect(img1.classes('hidden-pokemon')).toBe(true)
    expect(img1.attributes('src')).toBe(`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg`)
  })

  test('Debe de mostrar el pokemon si showPokemon = true', () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 100,
        showPokemon: true
      }
    })
    const img1 = wrapper.find('img')
    expect(img1.exists()).toBeTruthy()
    expect(img1.classes('fade-in')).toBe(true)
  })
})