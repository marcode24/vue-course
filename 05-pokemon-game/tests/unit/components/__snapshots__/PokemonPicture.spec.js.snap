// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPicture component Debe hacer match con el snapshot 1`] = `<div class="pokemon-container"><img class="hidden-pokemon" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/12.svg" alt="pokemon-image"></div>`;
